pallets/asset-index/src/traits.rs

```diff
@@ -28,7 +28,14 @@ pub trait AssetRecorder<AccountId, AssetId, Balance> {
         availability: AssetAvailability,
     ) -> Option<AssetAvailability>;
 
-    fn remove_asset(id: &AssetId) -> DispatchResult;
+    /// Dispatches transfer to move assets out of the index’s account, if a liquid asset is specified, burns the SAFT otherwise.
+    /// Updates the index by burning the given amount of index token.
+    fn remove_asset(
+        id: AssetId,
+        units: Balance,
+        nav: Balance,
+        recipient: Option<AccountId>,
+    ) -> DispatchResult;
```

Comment:
`saft_registry` also have function `remove_saft` calling this method which terms to remove the total SAFT, should we separate this into two functions or change the `remove_saft` in `saft-registry` ?

Reply:
That'd be definitely an improvement to split this in to functions